packages/medusa/src/api/routes/admin/products/update-option.js

```diff
@@ -0,0 +1,48 @@
+import { MedusaError, Validator } from "medusa-core-utils"
+
+export default async (req, res) => {
+  const { id, optionId } = req.params
+
+  const schema = Validator.object().keys({
+    title: Validator.string(),
+    values: Validator.array().items(),
+  })
+
+  const { value, error } = schema.validate(req.body)
+  if (error) {
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
+  }
+
+  try {
+    const productService = req.scope.resolve("productService")
+    const product = await productService.retrieve(id)
+
+    if (value.title) {
+      await productService.updateOption(product._id, optionId, {
+        title: value.title,
+      })
+    }
+
+    if (value.values) {
+      await productService.updateOption(product._id, optionId, {
+        values: value.values,
+      })
+    }
```

Comment:
Don't need two calls here

Reply:
Not sure how to combine this without checking multiple combinations in `if () {} else if` etc..

Reply:
What's wrong with `await productService.updateOption(product._id, optionId, value)`?

Reply:
Ah I see... I had my head elsewhere - my bad. (late night sesh)

Reply:
Actually when looking at the ProductService it seems that you can only update the title

Reply:
The values are being updated, but we aren't really handling them at all right?. I will look into this.

Reply:
Now that I think of it, nothing really needs to be handled in regards to the values. Simply adding the following line `update["options.$.values"] = values`, should do the work.

Reply:
Does it even make sense to allow value updates? The values are stored on the ProductVariants so the values array would just be an aggregate of what is held in the product's variants. I think we should remove it altogether and let values live in the ProductVariant objects only

Reply:
I actually think, we spoke about this at some point in the past. I remember we chose to keep it for some reason, so maybe we should keep as it is for now, and then look at it at a later point? But I will remove the values update again, so only title can be touched.